### ISSUE TITLE
Support props in ESLint-based rules

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#1100]

**Comment by vilchik-elena** [comment]
```js
class A {
   static foo = 42; // fails with "Unexpected token = "
}
```

`espree` fails to parse it, but babel-eslint does

**Comment by simonedavico** [comment]
Wouldn't this be as simple as always using `babel-eslint` for parsing? Would you accept a PR?

**Comment by vilchik-elena** [comment]
@simonedavico good point, but are you sure there is no such syntax supported by espree and not supported by babel?

**Comment by simonedavico** [comment]
Unfortunately I am not an expert in js/es parsers...for sure, `babel`supports syntax up to `stage-3`, while `espree` does not. This is an issue at my company because we are using arrow functions as class properties, which are supported by `babel` but not `espree`. They are also shipped with hugely popular tools (as in [create react app](https://github.com/facebook/create-react-app)), so I think providing support is for the best.

I opened PR #1143 , would you be so kind to take a look and give some feedback?

**Comment by rstrahl** [comment]
We're encountering the same issue in our project, and its halted usage on front-end analysis for us.
As far as I can see in the tooling implementation tracking issue reports as of this writing, espree does not support class fields.  A quick check in the espree issue/PR tracker appears to support this.

Ref: https://github.com/tc39/proposal-class-fields/issues/57

**Comment by simonedavico** [comment]
@gersongoulart I actually wanted to make it configurable, but it would be more difficult to implement tests for each parser, since they are not feature equivalent. Of course with additional effort it could be achieved.

**Comment by slikts** [comment]
It would be nice if SonarJS would be able to reuse the ESLint configuration regarding which parser to use (since SonarJS uses ESLint's Espree anyway, so there's obvious overlap) and which files to include or exclude. It would make using SonarJS more streamlined and avoid having to duplicate settings, on top of supporting proposed JS featured.
